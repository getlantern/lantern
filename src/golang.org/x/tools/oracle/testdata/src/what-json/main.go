package main

// Tests of 'what' queries, -format=json.
// See go.tools/oracle/oracle_test.go for explanation.
// See what-json.golden for expected query results.

func main() {
	f() // @what call "f"
}
