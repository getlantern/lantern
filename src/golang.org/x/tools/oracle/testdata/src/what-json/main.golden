-------- @what call --------
{
	"mode": "what",
	"what": {
		"enclosing": [
			{
				"desc": "identifier",
				"start": 179,
				"end": 180
			},
			{
				"desc": "function call (or conversion)",
				"start": 179,
				"end": 182
			},
			{
				"desc": "expression statement",
				"start": 179,
				"end": 182
			},
			{
				"desc": "block",
				"start": 176,
				"end": 202
			},
			{
				"desc": "function declaration",
				"start": 164,
				"end": 202
			},
			{
				"desc": "source file",
				"start": 0,
				"end": 202
			}
		],
		"modes": [
			"callees",
			"callers",
			"callstack",
			"definition",
			"describe",
			"freevars",
			"implements",
			"pointsto",
			"referrers"
		],
		"srcdir": "testdata/src",
		"importpath": "what-json"
	}
}
